### PR TITLE
TSK-1331: allowed multiple sortBy declarations in REST api

### PR DESCRIPTION
**Pull request by mustaphazorgati**
I couldn't find a spot in our `rest-api` where query parameters were documented.
Since I don't have the time to document all query parameters I didn't document the new behaviour.

https://sonarcloud.io/dashboard?branch=task%2F1331&id=mustaphazorgati_taskana
Code Coverage: I have no idea why it dropped. I tested the addition of new code. 
Probably because the other methods I refactored were not tested in the first place

Code Smells: 

- [code smell 1](https://sonarcloud.io/project/issues?branch=task%2F1331&id=mustaphazorgati_taskana&open=AXM1aKBdrgK0Yoghlb9G&resolved=false&types=CODE_SMELL ): Was there before, won't fix

- [code smell 2](https://sonarcloud.io/project/issues?branch=task%2F1331&id=mustaphazorgati_taskana&open=AXM1aKDfrgK0Yoghlb9H&resolved=false&types=CODE_SMELL): Will be fixed shortly (whenever travis runs again)

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.MultiValueMap;
+
+import pro.taskana.common.api.BaseQuery.SortDirection;
+import pro.taskana.common.api.exceptions.InvalidArgumentException;
+import pro.taskana.common.internal.util.CheckedBiConsumer;
+
+public class QueryHelper {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java:12]: I don't like this name. 
Any suggestion? 
I'd love to expand this class in the future by adding e.g. the extraction of paging here, etc.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package pro.taskana.common.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java:1]: This package ok?

File: history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/simplehistory/rest/TaskHistoryEventController.java [hunk 3]
```diff
@@ -152,7 +109,7 @@ public TaskHistoryEventController(
     }
 
     HistoryQuery query = simpleHistoryService.createHistoryQuery();
-    query = applySortingParams(query, params);
```
Comment (holgerhagen) [inline history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/simplehistory/rest/TaskHistoryEventController.java:155]: According to clean code it is a good practice to return the changed value, even if you don't have to do it. What do you think?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/classification/rest/ClassificationController.java [hunk 4]
```diff
@@ -48,56 +49,33 @@
...
   private static final String CUSTOM_7_LIKE = "custom-7-like";
-
   private static final String CUSTOM_8_LIKE = "custom-8-like";
 
-  private static final String SORT_BY = "sort-by";
-
-  private static final String SORT_DIRECTION = "order";
-
   private final ClassificationService classificationService;
+  private final ClassificationRepresentationModelAssembler modelAssembler;
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/classification/rest/ClassificationController.java:68]: Do we have a rule in the coding guideline with respect to naming variables?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java (hunk 2 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java:1]: This package is part of the public API, isn't it? I don't think that the QueryHelper belongs to the public API.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java (hunk 1 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java:12]: What's wrong with the name? Maybe we should discuss this if you have concerns.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    if (allOrderBy != null && !allOrderBy.isEmpty() && "desc".equalsIgnoreCase(allOrderBy.get(i))) {
+      sortDirection = SortDirection.DESCENDING;
+    }
+    return sortDirection;
+  }
+
+  private static void verifyNotOnlyOrderByExists(List<String> allSortBy, List<String> allOrderBy)
+      throws InvalidArgumentException {
+    if (allSortBy == null && allOrderBy != null) {
+      throw new InvalidArgumentException(
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java:60]: don't we have a default for order by?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          String.format(
+              "Only '%s' were provided. Please also provide '%s' parameter(s)",
+              ORDER_DIRECTION, SORT_BY));
+    }
+  }
+
+  private static void verifyAmountOfSortByAndOrderByMatches(
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/QueryHelper.java:67]: see above

File: rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskCommentController.java [hunk 7]
```diff
@@ -201,45 +201,35 @@
...
-        case (CREATED):
-          if (params.getFirst(SORT_DIRECTION) != null
-              && "desc".equals(params.getFirst(SORT_DIRECTION))) {
-            taskComments.sort(Comparator.comparing(TaskComment::getCreated).reversed());
-          } else {
-            taskComments.sort(Comparator.comparing(TaskComment::getCreated));
+    QueryHelper.applyAndRemoveSortingParams(
+        params,
+        (sortBy, sortDirection) -> {
+          Comparator<TaskComment> comparator;
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskCommentController.java:214]: Don't we sort on the dababase?

File: lib/taskana-core/src/main/java/pro/taskana/common/internal/util/CheckedBiConsumer.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package pro.taskana.common.internal.util;
+
+@FunctionalInterface
+public interface CheckedBiConsumer<T, U, E extends Throwable> {
```
Comment (holgerhagen) [inline lib/taskana-core/src/main/java/pro/taskana/common/internal/util/CheckedBiConsumer.java:4]: What is the meaning of this name?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/classification/rest/ClassificationController.java (hunk 4 above)
Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/classification/rest/ClassificationController.java:68]: cop?

**Pull request merged**